### PR TITLE
docs: add workaround for TS custom commands

### PR DESCRIPTION
**Pull request by marmaladebacon**
closes [#27641](https://github.com/cypress-io/cypress/issues/27641)

### CODE REVIEW

**Comment by nagash77** [comment]
Hi @marmaladebacon you will need to sign the CLA before this PR can be merged.

File: docs/api/cypress-api/custom-commands.mdx [hunk 1]
```diff
@@ -897,6 +897,31 @@ IntelliSense to show helpful documentation. See the
 [`cypress-example-todomvc`](https://github.com/cypress-io/cypress-example-todomvc#cypress-intellisense)
 repository for a working example.
 
+#### 6. Create a function that adds the custom command
+
+From cypress@12.17.4, if you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:902]: The "@" actually creates a `mailto:` link in the markdown.

```suggestion
From version [12.17.4](/guides/references/changelog#12-17-4), if you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
```

File: docs/api/cypress-api/custom-commands.mdx [hunk 2]
```diff
@@ -897,6 +897,31 @@ IntelliSense to show helpful documentation. See the
...
+```typescript title="/cypress/support/commands.ts"
+export function registerCommands(){
+  Cypress.Commands.add("login", (email, password) => { ... })
+}
+```
+
+```typescript title="/cypress/support/e2e.ts"
+import { registerCommands } from './commands'
+registerCommands()
+```
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:923]: The "@" actually creates a `mailto:` link in the markdown. Here are some edits to give more context

```suggestion
Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.

If you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:

```typescript
// cypress/support/commands.ts

Cypress.Commands.add("login", (email, password) => { ... })
```

```typescript
// cypress/support/e2e.ts

import './commands'
```

To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.

```typescript
// cypress/support/commands.ts

export function registerCommands(){
  Cypress.Commands.add("login", (email, password) => { ... })
}
```

```typescript
// cypress/support/e2e.ts

import { registerCommands } from './commands'
registerCommands()
```
```

File: docs/api/cypress-api/custom-commands.mdx [hunk 3]
```diff
@@ -897,6 +897,33 @@ IntelliSense to show helpful documentation. See the
...
 
+#### 6. Create a function that adds the custom command
+
+Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.
+
+If you are using TypeScript and have `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:
+
+```typescript title="/cypress/support/commands.ts"
+Cypress.Commands.add("login", (email, password) => { ... })
+```
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:908]: Weird that the markdown is breaking in the suggestion. Here are the rest of the original batch, plus a couple tiny improvements. 

```suggestion
Cypress [12.17.4](/guides/references/changelog#12-17-4) includes a Webpack upgrade (v4 to v5), which [tree shakes out](https://webpack.js.org/blog/2020-10-10-webpack-5-release/#inner-module-tree-shaking) any side-effects or files that only include side-effects.

If you are using TypeScript and have Webpack's `sideEffects:false` set in your package.json, custom Cypress commands will not be registered by the common pattern of writing the commands in one file and having them be run as a side effect of importing the file. For example:

```typescript
// cypress/support/commands.ts

Cypress.Commands.add("login", (email, password) => { ... })
```

File: docs/api/cypress-api/custom-commands.mdx [hunk 4]
```diff
@@ -897,6 +897,33 @@ IntelliSense to show helpful documentation. See the
...
+
+```typescript title="/cypress/support/e2e.ts"
+import './commands'
+```
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:912]: Weird that the markdown is breaking in the suggestion. Here are the rest of the original batch, plus a couple tiny improvements. 

```suggestion

```
// cypress/support/e2e.ts

import './commands'
```

File: docs/api/cypress-api/custom-commands.mdx [hunk 5]
```diff
@@ -897,6 +897,33 @@ IntelliSense to show helpful documentation. See the
...
+
+To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.
+
+```typescript title="/cypress/support/commands.ts"
+export function registerCommands(){
+  Cypress.Commands.add("login", (email, password) => { ... })
+}
+```
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:920]:
```suggestion
To support `sideEffects:false`, you can wrap the Cypress commands in a function that will be imported by the support file.

```typescript
// cypress/support/commands.ts

export function registerCommands(){
  Cypress.Commands.add("login", (email, password) => { ... })
}
```

```

File: docs/api/cypress-api/custom-commands.mdx [hunk 6]
```diff
@@ -897,6 +897,33 @@ IntelliSense to show helpful documentation. See the
...
+
+```typescript title="/cypress/support/e2e.ts"
+import { registerCommands } from './commands'
+registerCommands()
+```
```
Comment (jaffrepaul) [inline docs/api/cypress-api/custom-commands.mdx:925]:
```suggestion
```typescript
// cypress/support/e2e.ts

import { registerCommands } from './commands'
registerCommands()
```

**Comment by jaffrepaul** [comment]
Thank you for the contribution @marmaladebacon This was getting crazier and crazier with the formatting. Instead of wrestling with the markdown issues in the GH web UI, I just ported this over to [this PR](https://github.com/cypress-io/cypress-documentation/pull/5510) and merged it. Much appreciated!

**Pull request closed without merging**